### PR TITLE
gazebo_ros_camera: Added accessor methods for camera properties

### PR DESCRIPTION
**Pull request by kjeppesen1**
Added accessor methods to gazebo_ros_camera for subclass access to camera properties.

### CODE REVIEW

**Comment by kjeppesen1** [comment]
@scpeters @jacobperron let me know if you have any questions or suggestions

File: gazebo_plugins/src/gazebo_ros_camera.cpp [hunk 1]
```diff
@@ -838,6 +838,26 @@ void GazeboRosCamera::OnTrigger(const std_msgs::msg::Empty::SharedPtr)
...
 }
 
+const std::vector<gazebo::rendering::CameraPtr> GazeboRosCamera::GetCameras() const
+{
+  return impl_->camera_;
+}
+
+const gazebo_ros::Node::SharedPtr GazeboRosCamera::GetRosNode() const
+{
+  return impl_->ros_node_;
```
Comment (scpeters) [inline gazebo_plugins/src/gazebo_ros_camera.cpp]: just curious, why do you need the ROS node pointer? are you planning to create extra publishers?

Comment (kjeppesen1) [reply gazebo_plugins/src/gazebo_ros_camera.cpp]: Yes my subclass creates a subscriber to toggle the lens caps. Do you have a better solution in mind?

Comment (jacobperron) [reply gazebo_plugins/src/gazebo_ros_camera.cpp]: I'm not sure if it works, but you might be able to call `gazebo_ros::Node::Get(_sdf);` in the subclasses `Load` method. Then you'll also be able to parse any custom SDF tags specific to your plugin.

Comment (kjeppesen1) [reply gazebo_plugins/src/gazebo_ros_camera.cpp]: @scpeters @jacobperron Thanks for the suggestion. I can confirm that `gazebo_ros::Node::Get(_sdf);` does work. I have updated the PR to remove the ros node accessor.

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp [hunk 2]
```diff
@@ -162,6 +162,15 @@ class GazeboRosCamera
   /// param[in] _enabled True to enable.
   void SetCameraEnabled(const bool _enabled);
 
+  // Get pointer to the cameras
+  const std::vector<gazebo::rendering::CameraPtr> GetCameras() const;
+
+  // Get camera name
+  const std::string GetCameraName() const;
```
Comment (jacobperron) [inline gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp]: What advantage is there in returning const values here? I guess returning const references might save on some copying, but I think it's fine to return non-const values.

Comment (kjeppesen1) [reply gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp]: That was to convey the design intent that these values are not meant to modified, but rather just be retrieved for reference. Let me know if you see no value in that and I can delete that specifier.

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp [hunk 3]
```diff
@@ -162,6 +162,15 @@ class GazeboRosCamera
   /// param[in] _enabled True to enable.
   void SetCameraEnabled(const bool _enabled);
 
+  // Get pointer to the cameras
+  const std::vector<gazebo::rendering::CameraPtr> GetCameras() const;
```
Comment (scpeters) [inline gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp]: cpplint wants an `#include <vector>` at the top of this file due to this addition

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp (hunk 2 above)
Comment (jacobperron) [reply gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp]: IIUC, these are being returned by value, so there's nothing wrong with (or preventing) the caller modifying their local copy.

Also, I think it's potentially more efficient to remove the const so the compiler can ~~treat the return value as an rvalue~~ take advantage of move semantics.

Comment (scpeters) [reply gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp]: I think he meant that you could return `const std::string&`?

Comment (jacobperron) [reply gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp]: After re-educating myself about move semantics, I think it's generally considered a bad idea to return reference types. I would simply remove the const from the return types.

Comment (kjeppesen1) [reply gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp]: Removed const from the return types

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp (hunk 3 above)
Comment (kjeppesen1) [reply gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp]: Added

**Review by jacobperron — APPROVED** [review]

**Pull request merged**